src/redux/slices/commentsSlice.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-param-reassign */
+import {
+  ActionReducerMapBuilder,
+  PayloadAction,
+  createAsyncThunk,
+  createSlice,
+} from '@reduxjs/toolkit';
+import { Comment, CommentData } from '../../types/Comment';
+import {
+  createComment,
+  getPostComments,
+  deleteComment,
+} from '../../api/comments';
+
+type CommentsState = {
+  items: Comment[];
+  loaded: boolean;
+  created: boolean;
+  hasError: boolean;
+};
+
+type NewComment = {
+  postId: number;
+  data: CommentData;
+};
+
+type CommentThunk = 'fetchComments' | 'addComment' | 'removeComment';
+
+const initialState: CommentsState = {
+  items: [],
+  loaded: false,
+  created: false,
+  hasError: false,
+};
+
+export const thunks = {
+  fetchComments: createAsyncThunk(
+    'comments/fetch',
+    (postId: number) => getPostComments(postId),
+  ),
+
+  addComment: createAsyncThunk(
+    'comments/add',
+    ({ postId, data }: NewComment) => createComment(postId, data),
+  ),
+
+  removeComment: createAsyncThunk(
+    'comments/delete',
+    (commentId: number, { dispatch }) => {
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      dispatch(removeCommentById(commentId));
+
+      return deleteComment(commentId);
+    },
+  ),
+};
+
+const commentsExtraReducer = (
+  builder: ActionReducerMapBuilder<CommentsState>,
+  thunkType: CommentThunk,
+) => {
+  const commentThunk = thunks[thunkType];
+
+  if (!commentThunk) {
+    throw new Error('Something went wrong');
+  }
+
+  builder
+    .addCase(commentThunk.pending, (state) => {
+      state.hasError = false;
+      state.created = false;
+    })
+    .addCase(commentThunk.fulfilled, (state, action) => {
+      switch (thunkType) {
+        case 'fetchComments':
```

Comment:
why do we need to check thunkType?

could you share reference from docs/article why you do like this? or add `addCase` for each statement